Improve reliability of tests

olivierdagenais opened the pull request:
This will handle the case where it's not _quite_ the same .NET framework
running, plus any difference will be identified by its line number.

olivierdagenais committed:
Add + use AssertGeneratedLinesEqual() method.

This will handle the case where it's not _quite_ the same .NET framework
running, plus any difference will be identified by its line number.

JefStat commented on Solink.AddIn.Helpers.Test/RestartableAddInGeneratorTest.cs:
```diff
@@ -83,7 +83,41 @@ private static void AssertGeneratedCode(string expected, CodeCompileUnit compile
+                while (el != null && al != null)
+                {
+                    checkLine();
+                    el = er.ReadLine(); al = ar.ReadLine();
```
`checkLine` and this line call readline. In essence your checking every second line now in this loop. Also line number is not incremented.

JefStat commented on Solink.AddIn.Helpers.Test/RestartableAddInGeneratorTest.cs:
```diff
@@ -83,7 +83,41 @@ private static void AssertGeneratedCode(string expected, CodeCompileUnit compile
+                checkLine();                                                 // </auto-generated>
+                checkLine();                                                 //-----------------
+                string el = er.ReadLine(), al = ar.ReadLine();
+                while (el != null && al != null)
```
`||` would be sufficient here and a slightly faster for short circuit evaluation.

JefStat commented on Solink.AddIn.Helpers.Test/RestartableAddInGeneratorTest.cs:
```diff
@@ -83,7 +83,41 @@ private static void AssertGeneratedCode(string expected, CodeCompileUnit compile
+                checkLine();                                                 //     the code is regenerated.
+                checkLine();                                                 // </auto-generated>
+                checkLine();                                                 //-----------------
+                string el = er.ReadLine(), al = ar.ReadLine();
```
Another problem with reading the line and not asserting the values before reading the next line. Also line number is not incremented.

olivierdagenais committed:
TDD: every second line is skipped!

When I introduced checkLine(), I forgot to remove calls to ReadLine()
inside the while loop.

olivierdagenais committed:
Add some tests for AssertGeneratedLinesEqual().

olivierdagenais committed:
Add edge tests for AssertGeneratedLinesEqual().

olivierdagenais commented:
I fixed the embarrassing oversights and added more tests.
